src/common.fypp

```diff
@@ -156,4 +156,59 @@ ${prefix + joinstr.join([line.strip() for line in txt.split("\n")]) + suffix}$
   #:endcall
 #:enddef
 
+#!
+#! Generates an array rank suffix for subarrays along a dimension
+#!
+#! Args:
+#!   rank (int): Rank of the original variable
+#!   dim (int): Dimension of the variable
+#!
+#! Returns:
+#!   Array rank suffix string enclosed in braces
+#!
+#! E.g.,
+#!  select_subvector(5, 2)
+#!   -> (j_, :, j___, j____, j_____)
+#!
```

Comment:
Nice additions here (albeit I hope in future Fortran can offer a better way to do this!).

Reply:
Is there any value in providing an optional argument to make `j` some other letter? Not sure if situations would arise where that was useful? Although in any case, if you don't need it now, this does not have to be done.

Reply:
Thank you for the suggestion. done.